.github/workflows/test-and-build-workflow.yml

```diff
@@ -30,20 +30,20 @@ jobs:
         uses: actions/checkout@v2
         with:
           repository: 'opensearch-project/OpenSearch'
-          ref: '1.0.0-beta1'
+          ref: '1.x'
           path: OpenSearch
       - name: Build OpenSearch
         working-directory: ./OpenSearch
-        run: ./gradlew publishToMavenLocal -Dbuild.version_qualifier=beta1 -Dbuild.snapshot=false
```

Comment:
From the changes, I see we are leveraging `fallbackSetting` to take care of legacy settings. 
Ref: https://github.com/opensearch-project/OpenSearch/blob/0ba0e7cc26060f964fcbf6ee45bae53b3a9941d0/server/src/main/java/org/opensearch/common/settings/Setting.java#L1703:38

Moving to 1.x might not be needed. 
But eventually we will have to build out of 1.x. 

Let me know your thoughts on this, I propose:
1. cut a branch for `1.0.0.0-beta1` for all the plugins.
2. `1.0.0.0-beta1` will still build out of OpenSearch `1.0.0-beta1`, this helps having stable builds.
3. `main` will build out of the latest feature branch on OpenSearch, in this case `1.x`, this helps absorbing changes from dependencies sooner rather than later.